src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,83 @@
+package duke;
+
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.util.ArrayList;
+
+/**
+ * Represents the storage space, together with
+ * functions to read and write data from the file path.
+ */
+public class Storage {
+    /**
+     * The filePath where data is stored.
+     */
+    private String filePath;
+
+    /**
+     * Constructor for Storage.
+     *
+     * @param filePath location where the file is to be stored.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Returns the list of tasks stored if the file already
+     * exits or else returns an empty list.
+     *
+     * @return list containing saved tasks or empty list.
+     * @throws Exception If an unexpected error occurs.
+     */
+    public ArrayList<Task> readData() throws Exception {
+        File file = new File(filePath);
+        ArrayList<Task> list = new ArrayList<>();
+
+        if (file.exists()) {
+
+            BufferedReader r = new BufferedReader(new FileReader(file));
+            String t = r.readLine();
+
+            while (t != null) {
+                char i = t.charAt(0);
+                if (i == 'T') {
+                    String[] in = t.split(":");
+                    list.add(new Todo(in[2]));
+                } else if (i == 'D') {
+                    String[] in = t.split(":");
+                    list.add(new Deadline(in[2], in[3]));
+                } else if (i == 'E') {
+                    String[] in = t.split(":");
+                    list.add(new Event(in[2], in[3]));
+                } else {
+                    // do nothing
+                }
+                t = r.readLine();
+            }
+        }
```

Comment:
A neat set of code!

However, I was wondering if you could reformat this? It follows an arrow style where code is indented outward.

Nonetheless, Good job!! Very intuitive!